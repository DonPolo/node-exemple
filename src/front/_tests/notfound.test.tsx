import React from 'react';
import { mount } from 'enzyme';
import { BrowserRouter } from 'react-router-dom';
import NotFound from '../pages/notfound.page';
import fetch from './_mocks/fetch.mock';
import { SinonStub } from 'sinon';

describe('Test Analytics page', () => {
  let fakeFetch: SinonStub;
  beforeAll(() => {
    fakeFetch = fetch.mock();
  });
  test('Render the page correctly', async () => {
    const page = mount(
      <BrowserRouter>
        <NotFound />
      </BrowserRouter>,
    );
    expect(page).toMatchSnapshot();
  });
  afterAll(() => {
    fetch.unmock(fakeFetch);
  });
});
