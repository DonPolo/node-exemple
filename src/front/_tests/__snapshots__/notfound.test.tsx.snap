// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Analytics page Render the page correctly 1`] = `
<BrowserRouter>
  <Router
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "createHref": [Function],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "pathname": "/",
          "search": "",
          "state": undefined,
        },
        "push": [Function],
        "replace": [Function],
      }
    }
  >
    <Page>
      <div
        className="error"
      >
        <img
          alt="logo"
          src="/pic/lifee.png"
        />
        <h1>
          404
        </h1>
        <h2>
          Oooooups this page doesn't exist !
        </h2>
      </div>
    </Page>
  </Router>
</BrowserRouter>
`;
